core/cloudflow-streamlets/src/main/scala/cloudflow/streamlets/bytearray/ByteArrayUtil.scala

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright (C) 2016-2020 Lightbend Inc. <https://www.lightbend.com>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package cloudflow.streamlets.bytearray
+
+import java.security.MessageDigest
+import java.util.Base64
+
+import cloudflow.streamlets.SchemaDefinition
+
+object ByteArrayUtil {
+  val Format     = "bytearray"
+  val schemaName = "ByteArraySchema"
+
+  def createSchemaDefinition() = SchemaDefinition(
```

Comment:
It might be interesting to add a schema-name to ByteArrayInlet / Outlet, to get like at least a 'nominal type' for byte arrays.
Doing it like here means any byte array inlet and byte array outlet can read  / write to a topic even if they do not match at all, which seems like a programming error waiting to happen.

Reply:
Have you tested this for instance with a byte array inlet and an avro outlet connected  to the same topic? I think you will get a verification error in the blueprint. See `checkCompatibility` in `cloudflow.blueprint.Topic`

Reply:
Of course it will fail. The idea is to use bytearray as inlet/outlet on the topics, where non cloudflow app writing/reading

Reply:
Ok so this is then only useful for external inlets and outlets. It's probably handy, also for verification, to know that these are in fact external inlets and outlets.

Reply:
renamed inlet outlet and reworked documentation accordingly